src/page-modules/assistant/server/trip-cache.ts

```diff
@@ -0,0 +1,11 @@
+import TTLCache from '@isaacs/ttlcache';
+
+let tripCache: TTLCache<unknown, unknown> | null = null;
+
+export function getCache(): TTLCache<unknown, unknown> {
+  if (!tripCache) {
+    tripCache = new TTLCache({ ttl: 120000 });
+  }
+
+  return tripCache;
+}
```

Comment:
Is this the approach you would go for, @mikaelbr ? Importing this wherever it's needed. Seems that the cache object would be recreated at times, such as when navigating to a new site, but not sure if that is a problem. Another approach I was thinking of is to create this object in the `/src/page-modules/assistant/server/journey-planner/index.ts` file and the `getServerSideProps` fetch from the cache using a (new) endpoint.

Reply:
Not what you are asking, but I think trip-cache possibly should have functions for explicit caching. So instead of having this at consumer site:

```ts
if (tripCache.has(`/api/assistant/trip?${queryString}`)) {
        return tripCache.get(`/api/assistant/trip?${queryString}`) as TripData;
      }
```

You would have

```ts
const potentiallCached = getAssistantTripIfCached(query);
```

And thus hide the cache key generation and avoid error source where you don't match proper cache keys.

Reply:
Also I think maybe what you are saying about placing it in `getServerSideProps` is the approach I would go for.

Reply:
Good idea. And then it would make sense to create it as a module I think